### ISSUE TITLE
How to clear earlier async errors?

### ISSUE THREAD

**Issue by magnattic** [MrWolfZ/ngrx-forms#143]
This is my use case:
1. User inputs text into field, async validation triggers and fails, setting the async validation error.
2. User clears text, async validation errors should be cleared and only sync validation "required" error should be set.

I tried firing the ClearAsyncErrorAction when the field value is changing to empty, but the errors still are there. I can see the action in the ReduxDevTools and the values in it seem correct.

How else would you reset the async validation errors when the input value changes to something that is clearly invalid?

This is my effect code:

```

  @Effect() loadCities$ = merge(
    this.actions$.pipe(
      ofType(ROUTER_NAVIGATION),
      first(),
      withLatestFrom(this.store.select(fromChange.getProductPageFormLocation)),
      map(([_action, location]) => location.controls.postalCode)
    ),
    this.store.select(fromChange.getProductPageFormPostalCode)
  ).pipe(
    distinctUntilChanged((a, b) => a.value === b.value),
    // filter(postalCodeInput => postalCodeInput.value != null && postalCodeInput.value.length === 5),
    withLatestFrom(this.store.select(fromChange.getProductPageFormLocation)),
    switchMap(([postalCodeControl, locationControl]) => {
      if (postalCodeControl.value == null) {
        return of(new ClearAsyncErrorAction(postalCodeControl.id, 'postalCodeUnknown'));
      }
      return merge(
        of(new StartAsyncValidationAction(postalCodeControl.id, 'postalCodeUnknown')),
        of(new SetValueAction(locationControl.controls.city.id, '')),
        this.cityService.getCitiesForPostalCode(postalCodeControl.value).pipe(
          mergeMap(cities => [
            cities.length > 1 ? ApiActions.multipleCitiesFound({ cities: cities }) : ApiActions.cityFound({ city: cities[0] }),
            new ClearAsyncErrorAction(postalCodeControl.id, 'postalCodeUnknown')
          ]),
          catchError(error => of(new SetAsyncErrorAction(postalCodeControl.id, 'postalCodeUnknown', true)))
        )
      );
    })
```

In short: When postalCode.value is empty, I want to clear the remaining '$postalCodeUnknown' error and instead set the 'required' error.

**Comment by MrWolfZ** [comment]
Hmm, `ClearAsyncErrorAction` is the correct way of doing this and your code snippet looks correct to me as well. Can you maybe post the relevant parts of the redux dev tools output or create a reproduction in a [sandbox](https://codesandbox.io/s/angular)?

**Comment by magnattic** [comment]
I believe I found the problem in https://github.com/MrWolfZ/ngrx-forms/blob/master/src/group/reducer/clear-async-error.ts:

```
  if (state.pendingValidations.indexOf(action.name) < 0) {
    return state;
  }
```

Because of this code, async errors can only be cleared if there is a async validation pending. Is this as you intended it?

Because in my usecase I want to clear earlier async errors without triggering a new async validation with `StartAsyncValidationAction`.

**Comment by MrWolfZ** [comment]
You are absolutely right, thanks for pointing this out. I have just published version [4.0.2](https://github.com/MrWolfZ/ngrx-forms/releases/tag/v4.0.2), which fixes this issue.
